maven-api-impl/src/main/java/org/apache/maven/internal/impl/DefaultToolchainsXmlFactory.java

```diff
@@ -53,10 +52,6 @@ public PersistedToolchains read(@Nonnull XmlReaderRequest request) throws XmlRea
             throw new IllegalArgumentException("reader or inputStream must be non null");
         }
         try {
-            InputSource source = null;
-            if (request.getModelId() != null || request.getLocation() != null) {
-                source = new InputSource(request.getModelId(), request.getLocation());
-            }
```

Comment:
This is weird, source is never used? Was this found part of a new rule/detection?

Reply:
Yes new rule was detected it.
@gnodet - why is not used here ....?

Reply:
Toolchains do not currently fully support location tracking. I'll see if that can be done.